onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -302,9 +302,12 @@ QnnLog_Level_t QnnBackendManager::MapOrtSeverityToQNNLogLevel(logging::Severity
 }
 
 Status QnnBackendManager::ResetQnnLogLevel() {
-  auto ort_log_level = logger_->GetSeverity();
-  LOGS(*logger_, INFO) << "Reset Qnn log level to ORT Logger level: " << (unsigned int)ort_log_level;
-  return UpdateQnnLogLevel(ort_log_level);
+  if (logger_ != nullptr && backend_initialized_) {
+    auto ort_log_level = logger_->GetSeverity();
+    LOGS(*logger_, INFO) << "Reset Qnn log level to ORT Logger level: " << (unsigned int)ort_log_level;
```

Comment:
I think it's still not fully fixed. How to guarantee that the logger_ is always valid? Need a lock.

Reply:
I don't think it resolves the race condition here. This method is triggered from callback (a separate thread), and it accesses the logger_ and some Qnn resource. However the status of logger_ and Qnn resource can be changed at any time, even it passed the if (logger_ != nullptr && backend_initialized_).
For example, during session close

1. if (logger_ != nullptr && backend_initialized_) is true
2. QNN EP releases all resource
3. auto ort_log_level = logger_->GetSeverity(); -- not valid anymore, -> still crash

Reply:
You are correct @hectorsvc. Resolved